sonar-db/src/main/java/org/sonar/db/permission/PermissionTemplateDao.java

```diff
@@ -218,34 +218,51 @@ public void updatePermissionTemplate(Long templateId, String templateName, @Null
     }
   }
 
+  /**
+   * @deprecated since 5.2 use the method with dbSession instead
+   */
+  @Deprecated
   public void insertUserPermission(Long templateId, Long userId, String permission) {
+    DbSession session = myBatis.openSession(false);
+    try {
+      insertUserPermission(session, templateId, userId, permission);
+    } finally {
+      MyBatis.closeQuietly(session);
+    }
+  }
+
+  public void insertUserPermission(DbSession session, Long templateId, Long userId, String permission) {
     PermissionTemplateUserDto permissionTemplateUser = new PermissionTemplateUserDto()
       .setTemplateId(templateId)
       .setUserId(userId)
       .setPermission(permission)
       .setCreatedAt(now())
       .setUpdatedAt(now());
-    SqlSession session = myBatis.openSession(false);
+
+    mapper(session).insertUserPermission(permissionTemplateUser);
+    session.commit();
+  }
+
+  /**
+   * @deprecated since 5.2 please use method with DbSession
+   */
+  @Deprecated
+  public void deleteUserPermission(Long templateId, Long userId, String permission) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)